src/Microsoft.SqlTools.ServiceLayer/Connection/ListDatabaseRequestHandler.cs

```diff
@@ -139,18 +158,18 @@ protected override void SetResponse(ListDatabasesResponse response, DatabaseInfo
             response.Databases = results;
         }
 
-        protected override DatabaseInfo CreateItem(DbDataReader reader)
+        protected override DatabaseInfo CreateItem(SMO.Database database)
         {
             DatabaseInfo databaseInfo = new DatabaseInfo();
-            SetProperties(reader, databaseInfo);
+            SetProperties(database, databaseInfo);
             return databaseInfo;
         }
 
-        protected virtual void SetProperties(DbDataReader reader, DatabaseInfo databaseInfo)
+        protected virtual void SetProperties(SMO.Database database, DatabaseInfo databaseInfo)
         {
-            databaseInfo.Options[ListDatabasesRequestDatabaseProperties.Name] = reader["name"].ToString();
-            databaseInfo.Options[ListDatabasesRequestDatabaseProperties.State] = reader["state"].ToString();
-            databaseInfo.Options[ListDatabasesRequestDatabaseProperties.SizeInMB] = reader["size"].ToString();
+            databaseInfo.Options[ListDatabasesRequestDatabaseProperties.Name] = database.Name;
+            databaseInfo.Options[ListDatabasesRequestDatabaseProperties.State] = database.State.ToString();
```

Comment:
before accessing server.Databases collection, either:
1. Use server.SetDefaultInitFields(typeof(Database), new [] {nameof(`<property1>`), nameof(`<property2>`) etc}) 
or 
2. Use server.Databases.ClearAndInitialize(new [] {nameof(`<property1>`), nameof(`<property2>`) 

where the `<property>` names are the ones you access here. This will dramatically improve performance

Reply:
on further thought...we should be very careful about fetching any properties besides Name for Azure databases in particular.
Azure has the potential to create major headaches here, as fetching most properties of each database requires logging in to the database. Compounding the problem is that if the connection string was to "master", sys.databases returns all the databases on the server, even ones that may be paused or that the user doesn't have access to. So fetching any property besides Name has side effects like:
1. Waking up paused databases
2. Throwing SqlException when login fails to the database.
3. Taking a looong time to log in to low SLO databases etc

It may be better to defer such property fetches until they are actually needed.